### PR TITLE
Schedule view as List

### PR DESCRIPTION
**Pull request by not-SAINT**
Add schedule view as List. Current day - expanded if there are tasks.
image

### CODE REVIEW

File: client/src/components/Schedule/ListView.tsx [hunk 1]
```diff
@@ -1,17 +1,219 @@
...
 type Props = {
   data: CourseEvent[];
   timeZone: string;
 };
 
-export function ListView({ data, timeZone }: Props) {
+export const ListView = ({ data, timeZone }: Props): React.ReactElement => {
+  const [currentWeek, setCurrentWeek] = useState(0);
+  const [startWeekDate, setStartWeekDate] = useState('');
+  const [endWeekDate, setEndWeekDate] = useState('');
```
Comment (yuliaHope) [inline client/src/components/Schedule/ListView.tsx]: Looks like start and end date are used only in render. So there is no need to store them in state, let's calculate these data directly in render, but memoized version(using useMemo or useCallback for example) for optimizing performance.

File: client/src/components/Schedule/ListView.tsx [hunk 2]
```diff
@@ -1,17 +1,219 @@
...
+    setCurrentWeek(currentWeek - 1);
+  };
+
+  const handleClickForward = () => {
+    setCurrentWeek(currentWeek + 1);
+  };
+
+  const getCurrentDayKey = useMemo(() => {
+    const day = moment().day();
+    const currentDay = day ? day - 1 : 6;
```
Comment (yuliaHope) [inline client/src/components/Schedule/ListView.tsx]: What a magic number `6` is?

File: client/src/components/Schedule/ListView.tsx [hunk 3]
```diff
@@ -1,17 +1,219 @@
...
+    }
+
+    return `${WEEK[currentDay]}0`;
+  }, [data]);
+
   return (
-    <div>
-      <h2>List</h2>
-      Data length = {data.length}. Timezone = {timeZone}
+    <div className={'List'}>
```
Comment (yuliaHope) [inline client/src/components/Schedule/ListView.tsx]: Let's replace with `className="List"`

File: client/src/components/Schedule/ListView.tsx [hunk 4]
```diff
@@ -1,17 +1,219 @@
...
+  const weekMap = new Array(weekLength).fill([]);
+
+  events.forEach((event: CourseEvent) => {
+    let eventDay = moment(event.dateTime).tz(timeZone).day();
+
+    if (eventDay === 0) {
+      eventDay = 6;
+    } else {
+      eventDay -= 1;
+    }
```
Comment (yuliaHope) [inline client/src/components/Schedule/ListView.tsx:122]: The same question about magic numbers here)

File: client/src/components/Schedule/ListView.tsx [hunk 5]
```diff
@@ -1,17 +1,219 @@
...
+
+      dayMap.push(event);
+      weekMap[eventDay] = dayMap;
+    }
+  });
+
+  return weekMap;
+};
+
+const dayEvents = (events: CourseEvent[], timeZone: string) => {
```
Comment (yuliaHope) [inline client/src/components/Schedule/ListView.tsx]: dayEvents -> getDayEvents

File: client/src/components/Schedule/ListView.tsx [hunk 6]
```diff
@@ -1,17 +1,219 @@
...
+            <span>{name}</span>
+          </th>
+        </tr>
+        <style jsx>{tableStyles}</style>
+      </tbody>
+    );
+  });
+};
+
+const weekElements = (events: CourseEvent[], selectedWeek: number, timeZone: string) => {
```
Comment (yuliaHope) [inline client/src/components/Schedule/ListView.tsx]: the same here `getWeekElements`

File: client/src/components/Schedule/ListView.tsx (hunk 1 above)
Comment (not-SAINT) [reply client/src/components/Schedule/ListView.tsx]: done

File: client/src/components/Schedule/ListView.tsx (hunk 2 above)
Comment (not-SAINT) [reply client/src/components/Schedule/ListView.tsx]: fix)

File: client/src/components/Schedule/ListView.tsx (hunk 3 above)
Comment (not-SAINT) [reply client/src/components/Schedule/ListView.tsx]: fix

File: client/src/components/Schedule/ListView.tsx (hunk 4 above)
Comment (not-SAINT) [reply client/src/components/Schedule/ListView.tsx:122]: fix

File: client/src/components/Schedule/ListView.tsx (hunk 5 above)
Comment (not-SAINT) [reply client/src/components/Schedule/ListView.tsx]: fix

File: client/src/components/Schedule/ListView.tsx (hunk 6 above)
Comment (not-SAINT) [reply client/src/components/Schedule/ListView.tsx]: fix

**Review by yuliaHope — APPROVED** [review]

**Pull request merged**